src/parcels/_core/particleset.py

```diff
@@ -632,19 +618,29 @@ def _get_simulation_start_and_end_times(
             )
             raise ValueError(msg)
 
+        endtime = timedelta_to_float(endtime - time_interval.left)
+
+    start_time = _get_start_time(first_release_time, time_interval, sign_dt, runtime)
+
+    if isinstance(runtime, np.timedelta64):
+        runtime = timedelta_to_float(runtime)
+
+    if endtime is None:
+        endtime = start_time + sign_dt * runtime
+
     return start_time, endtime
 
 
 def _get_start_time(first_release_time, time_interval, sign_dt, runtime):
     if time_interval is None:
-        time_interval = TimeInterval(left=np.timedelta64(0, "s"), right=runtime)
+        time_interval = TimeInterval(np.timedelta64(0, "s"), right=np.timedelta64(int(runtime * 1e9), "ns"))
 
     if sign_dt == 1:
```

Comment:
Separate to this PR, something I want to work on is typechecking in Parcels (I think in cases like this it would make it a lot easier to reason around what the types are for these different parameters)

Reply:
Yep, I agree that more typechecking would be good